src/foam/nanos/auth/Address.js

```diff
@@ -223,10 +223,18 @@ foam.CLASS({
         {
           args: ['postalCode'],
           predicateFactory: function(e) {
-            return e.GT(
-              foam.mlang.StringLength.create({
-                arg1: foam.nanos.auth.Address.POSTAL_CODE
-              }), 0);
+            return e.OR(
+              e.EQ(foam.nanos.auth.Address.COUNTRY_ID, 'ZA'),
+              e.EQ(foam.nanos.auth.Address.COUNTRY_ID, 'TT'),
+              e.EQ(foam.nanos.auth.Address.COUNTRY_ID, 'MY'),
+              e.EQ(foam.nanos.auth.Address.COUNTRY_ID, 'MX'),
+              e.EQ(foam.nanos.auth.Address.COUNTRY_ID, 'JM'),
+              e.EQ(foam.nanos.auth.Address.COUNTRY_ID, 'LB'),
+              e.GT(
+                foam.mlang.StringLength.create({
+                  arg1: foam.nanos.auth.Address.POSTAL_CODE
+                }), 0)
+            )
```

Comment:
question! We already check the postal code validations on the each configured country. Then We need this here?

Reply:
yeah, they want 2 error message, one is if it is empty, one is invalid